test/bundling.js

```diff
@@ -0,0 +1,827 @@
+  });
+
+  it('emits errors when the api call fails', function(done) {
+    var executor = newExecutor(new gax.BundleOptions({'delayThreshold': 100}));
+    var callback = sinon.spy(function(err, resp) {
+      expect(err).to.be.an.instanceOf(Error);
+      if (callback.callCount == 2) {
+        done();
+      }
+    });
+    var emitter = schedule(
+        executor, failing, createSimple([1], 'id'), callback);
+    var emitter2 = schedule(
+        executor, failing, createSimple([2], 'id'), callback);
+
+    emitter.runNow();
+  });
+
+  describe('with events', function() {
+    var executor = newExecutor(new gax.BundleOptions({'delayThreshold': 100}));
+    var spyApi;
+
+    function timedAPI(request, callback) {
+      var canceled = false;
+      setTimeout(function() {
+        if (!canceled) {
+          callback(null, request);
+        }
+      }, 10);
+      var emitter = new eventemitter2.EventEmitter2();
+      emitter.cancel = function() {
+        canceled = true;
+      };
+      return emitter;
+    }
+
+    beforeEach(function() {
+      spyApi = sinon.spy(apiCall);
+    });
+
+    describe('with listeners', function() {
+      it('ignore runNow once it\'s finished', function(done) {
+        var callCount = 0;
+        var event1 = schedule(executor, spyApi, createSimple([1, 2], 'id'));
+        event1.on('data', function(resp) {
+          callCount++;
+          // make sure this callback is called only once.
+          expect(callCount).to.eq(1);
+          expect(resp.field1).to.deep.equal([1, 2]);
+          expect(spyApi.callCount).to.eq(1);
+
+          // event1.runNow() does nothing, even though event2 is scheduled
+          // with the same bundle id.
+          event1.runNow();
+          expect(spyApi.callCount).to.eq(1);
+
+          setTimeout(done, 0);
+        });
+        event1.runNow();
+      });
+
+      it('shouldn\'t block next event after cancellation', function(done) {
+        var event1 = schedule(executor, spyApi, createSimple([1, 2], 'id'));
+        event1
+            .on('data', function() { done(new Error('should not reach')); })
+            .on('error', function() { done(new Error('should not reach')); });
+        event1.cancel();
+
+        setTimeout(function() {
+          expect(spyApi.callCount).to.eq(0);
+
+          var event2 = schedule(executor, spyApi, createSimple([3, 4], 'id'));
+          event2.on('data', function(resp) {
+            expect(resp.field1).to.deep.equal([3, 4]);
+            expect(spyApi.callCount).to.eq(1);
+            setTimeout(done, 0);
+          });
+          event2.on('error', done);
+          event2.runNow();
+        }, 0);
+      });
+
+      it('distinguishes a running task and a scheduled task.', function(done) {
+        var event1 = schedule(executor, timedAPI, createSimple([1, 2], 'id'));
+        event1.on('data', function() {
+          // Wait for 20msec to verify that event2 does not finish accidentally.
+          setTimeout(done, 20);
```

Comment:
After reconsidering timedAPI, this does not have to wait for 20msecs actually, although it should wait for the next cycle of the event, such as setTimeout(done, 0);  Done, and added some comments here.